### PR TITLE
Add Synthea dataset

### PR DESCRIPTION
**Pull request by namsaraeva**
Adds:
- Synthea dataset (in older DSTU2 format, since it is much smaller than the recent R4 one). This dataset contains 1000 synthetic patient medical records. No reply from fhiry developers on the correct FHIR version yet.

Fixes:
- Some .zip files have different directory name when unpacked, so I added a fix to `_get_non_existing_files()` function to handle this case instead of throwing an error that the directory was not found. The fix looks for the last created file/dir (meaning a .zip file was unpacked and a directory was created) in our dataset directory and changes the returned path to it. 
- The fix leaves out hidden files that begin with `.`

Questions:
- Do we want more Synthea datasets? The COVID-19 one is available only in `.csv` format though. Other datasets are too heavy.

### CODE REVIEW

File: ehrapy/data/_datasets.py [hunk 1]
```diff
@@ -681,3 +682,40 @@ def heart_disease(
...
+        encoded: Whether to return an already encoded object.
+        columns_obs_only: Columns to include in obs only and not X.
+
+    Returns:
+        :class:`~anndata.AnnData` object of the 1K Sample Synthetic Patient Records Data Set
+
+    Example:
+        .. code-block:: python
+
+            import ehrapy as ep
```
Comment (Zethson) [inline ehrapy/data/_datasets.py:706]:
```suggestion
            >>> import ehrapy as ep
```

File: ehrapy/data/_datasets.py [hunk 2]
```diff
@@ -681,3 +682,40 @@ def heart_disease(
...
+
```
Comment (Zethson) [inline ehrapy/data/_datasets.py:707]:
```suggestion
```

File: ehrapy/data/_datasets.py [hunk 3]
```diff
@@ -681,3 +682,40 @@ def heart_disease(
...
+            adata = ep.dt.synthea_1k_sample(encoded=True)
```
Comment (Zethson) [inline ehrapy/data/_datasets.py:708]:
```suggestion
            >>> adata = ep.dt.synthea_1k_sample(encoded=True)
```

File: ehrapy/io/_read.py [hunk 4]
```diff
@@ -488,8 +488,14 @@ def _get_non_existing_files(file: Path, download_dataset_name: str, backup_url:
     output_path_name = download_default_name.replace(archive_extension, "") if is_archived else download_default_name
     output_file_or_dir = ehrapy_settings.datasetdir / output_path_name
     moved_path = Path(str(output_file_or_dir)[: str(output_file_or_dir).rfind("/") + 1]) / download_dataset_name
-    shutil.move(output_file_or_dir, moved_path)  # type: ignore
-    file = moved_path
+    if moved_path.exists():
+        shutil.move(output_file_or_dir, moved_path)  # type: ignore
+        file = moved_path
+    elif not moved_path.exists():
```
Comment (Zethson) [inline ehrapy/io/_read.py:494]: Can you add a comment above this to explain why we need this case? I know it but other people reading the code may not yet know the answer.

File: ehrapy/io/_read.py [hunk 5]
```diff
@@ -488,8 +488,14 @@ def _get_non_existing_files(file: Path, download_dataset_name: str, backup_url:
...
+        list_of_paths = [path for path in ehrapy_settings.datasetdir.glob("*/") if not path.name.startswith(".")]
+        latest_path = max(list_of_paths, key=lambda x: x.stat().st_ctime)
```
Comment (Zethson) [inline ehrapy/io/_read.py:496]: Rename `x`

**Review by Zethson — APPROVED** [review]

File: ehrapy/data/_datasets.py (hunk 3 above)
Comment (namsaraeva) [reply ehrapy/data/_datasets.py:708]: What do you mean by this? @Zethson

**Pull request merged**